OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -305,41 +457,123 @@ public string VoicePhraseForOrder(Actor self, Order order)
 				case "Enter":
 				case "ReturnToBase":
 				case "Stop":
-					return info.Voice;
+					return Info.Voice;
 				default: return null;
 			}
 		}
-	}
 
-	class AircraftMoveOrderTargeter : IOrderTargeter
-	{
-		public string OrderID { get { return "Move"; } }
-		public int OrderPriority { get { return 4; } }
-		public bool OverrideSelection { get { return false; } }
+		public void ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString == "Move")
+			{
+				var cell = self.World.Map.Clamp(order.TargetLocation);
+				var explored = self.Owner.Shroud.IsExplored(cell);
 
-		readonly AircraftInfo info;
+				if (!explored && !Info.MoveIntoShroud)
```

Comment:
Could you change this to `if (!Info.MoveIntoShroud && !self.Owner.Shroud.IsExplored(cell))` and drop the `var` above? Then we don't need the `IsExplored` lookup when `MoveIntoShroud` is `true`.